packages/grpc-js/src/resolver-dns.ts

```diff
@@ -302,6 +295,41 @@ class DnsResolver implements Resolver {
     }
   }
 
+  private async lookup(hostname: string): Promise<TcpSubchannelAddress[]> {
+    if (GRPC_NODE_USE_ALTERNATIVE_RESOLVER) {
+      const records = await Promise.allSettled([
```

Comment:
If both of these are rejected, the whole `lookup` function should reject. This should fix the problem with the test that expects lookup failures.

Reply:
Ahhh yes. Totally missed that. Good there is the test there 🙇